src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/RuntimeBroker.cs

```diff
@@ -312,6 +312,21 @@ private IntPtr GetParentWindow(CoreUIParent uiParent)
                         var errorMessage = "Could not acquire token silently.";
                         msalTokenResponse = WamAdapters.HandleResponse(result, authenticationRequestParameters, _logger, errorMessage);
                     }
+
+                    if (acquireTokenSilentParameters.ForceRefresh && !string.IsNullOrEmpty(msalTokenResponse.AccessToken))
```

Comment:
The AuthResult of recent version of native MSALs will have an attribute indicating the token source whose value can be "IdentityProvider" or "cache" etc.. Unless MsalRuntime also provide a similar mechanism, the only way to approximately know whether a token is new is to run the same token acquisition code path immediately and assert the new token will be different than the old token.

Reply:
You can look at how we test MSAL Runtime broker [here.](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/79f951a079ce2db3290874c7358d5c3242fbbe05/tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/RuntimeBrokerTests.cs#L174) As you can see, we only have the one `TokenSource.Broker` result that represents both from IDP and broker cache so there is no way to tell from that property if it is form the cache with that property. I suggest making a second silent call with the new api and verifying that the second access token does not match the first one as @rayluo suggests.

Reply:
@iulico-1 - here's my attempt at an integration test for this. It uses ROPC, because we don't run interactive auth in MSAL.NET

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4459/files

(feel free to merge it into your branch)

Reply:
Excellent. Thank you @bgavrilMS !